components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -2268,7 +2268,8 @@ def removeWrapperAndNullableTypes(types):
                 parentName = descriptor.getParentName()
                 while parentName:
                     descriptor = descriptorProvider.getDescriptor(parentName)
-                    extras += [descriptor.path, descriptor.bindingPath]
+                    if current_name != descriptor.ifaceName:
+                        extras += [descriptor.path, descriptor.bindingPath]
```

Comment:
This fixes a bug in binding generation - `AbortSignal` both implements `EventTarget` and is part of `AddEventListenerOptions` in `EventTarget`. This resulted in the binding generation trying to make `EventTarget` import itself.